### PR TITLE
Minor grammatical improvements to documentation writing guide

### PR DESCRIPTION
**Pull request by gabrieltaylor**

### CODE REVIEW

File: lib/elixir/pages/Writing Documentation.md [hunk 1]
```diff
@@ -53,33 +53,41 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
 
 ## Recommendations
 
-There are a couple tips we recommend developers to follow when writing documentation:
+When writing documentation:
 
-  * Keep the first paragraph of the documentation concise and simple, typically one-line. Tools like [ExDoc](https://github.com/elixir-lang/ex_doc/) use the first line to generate a summary.
+  * Keep the first paragraph of the documentation concise, typically one-line. Tools like [ExDoc](https://github.com/elixir-lang/ex_doc/) use the first line to generate a summary.
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: Do we need this change? I like the advice of keeping it both concise *and* simple, they're not the same thing but both important IMO.

File: lib/elixir/pages/Writing Documentation.md [hunk 2]
```diff
@@ -53,33 +53,41 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
 
 ## Recommendations
 
-There are a couple tips we recommend developers to follow when writing documentation:
+When writing documentation:
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: I liked the friendly wording of the previous version. Do we have a clear gain with this change? 😕

File: lib/elixir/pages/Writing Documentation.md [hunk 3]
```diff
@@ -53,33 +53,41 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
...
+  Markdown uses backticks (`` ` ``) to quote code. Elixir builds on top of that to automatically generate links when module or function names are referenced. For this reason, always use full module names. If you have a module called `MyApp.Hello`, always reference it as `` `MyApp.Hello` `` and never as `` `Hello` ``.
 
-  * When documenting a function with multiple clauses, the documentation must be placed before the first clause. Documentation is always per function and arity and not per clause.
+  * Reference functions by name and arity if they are local, as in `` `world/1` ``, or by module, name and arity if pointing to an external module: `` `MyApp.Hello.world/1` ``.
+
+  * Reference a `@callback` by prepending `c:`, as in `` `c:world/1` ``.
+
+  * Reference a `@type` by prepending `t:`, as in `` `t:values/0` ``.
+
+  * Start new sections with second level markdown headings `##`. The first heading is reserved for module and function names.
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: Probably worth saying "The first **level** headers are reserved for module and function names", wdyt?

File: lib/elixir/pages/Writing Documentation.md [hunk 4]
```diff
@@ -53,33 +53,41 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
...
+
+  * Place documentation before the first clause of multi-clause functions. Documentation is always per function and arity and not per clause.
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: Maybe "per function and arity" -> "per function", considering it's important Elixir/Erlang knowledge to know that arity is a defining part of a function? Not sure though.

File: lib/elixir/pages/Writing Documentation.md [hunk 5]
```diff
@@ -101,9 +109,9 @@ Similarly, developers can add `@doc false` to functions they do not want to be p
...
+However, keep in mind that adding `@doc false` does not make the function private. The function above can still be invoked as `MyApp.Sample.add(1, 2)`. Not only that, if `MyApp.Sample` is imported, the `add/2` function will also be imported into the caller. For those reasons, be cautious when adding `@doc false` to functions, instead either:
 
-  * Move the undocumented function to a module with `@moduledoc false`, like `MyApp.Hidden`, ensuring the function won't be accidentally exposed or imported. Remember you can use `@moduledoc false` to hide a whole module and still document each function with `@doc`. Tools will still ignore the module.
+  * Move the undocumented function to a module with `@moduledoc false`, like `MyApp.Hidden`, ensuring the function won't be accidentally exposed or imported. Remember you can use `@moduledoc false` to hide a whole module and still document each function with `@doc`. Tools will still ignore the module. Or,
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: We don't need the final "Or," here if we use capitalized initials in the bullet list. Let's keep just a trailing period here.

File: lib/elixir/pages/Writing Documentation.md [hunk 6]
```diff
@@ -53,33 +53,41 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
...
+
+  * Start new sections with second level markdown headings `##`. The first heading is reserved for module and function names.
+
+  * Place documentation before the first clause of multi-clause functions. Documentation is always per function and arity and not per clause.
 
 ## Doctests
 
-We recommend developers to include examples in their documentation, often under its own `## Examples` heading. To ensure examples do not get out of date, Elixir's test framework (ExUnit) provides a feature called doctests that allows developers to test the examples in their documentation. Doctests work by parsing out code samples starting with `iex>` from the documentation. You can read more about it at `ExUnit.DocTest`.
+We recommend developers include examples in their documentation, often under their own `## Examples` heading. To ensure examples do not get out of date, Elixir's test framework (ExUnit) provides a feature called doctests that allows developers to test the examples in their documentation. Doctests work by parsing out code samples starting with `iex>` from the documentation. You can read more about it at `ExUnit.DocTest`.
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: "recommend developers include" is correct over "recomment developers to include", but maybe "We recommend *that* developers include" reads even nicer?

File: lib/elixir/pages/Writing Documentation.md (hunk 5 above)
Comment (eksperimental) [inline lib/elixir/pages/Writing Documentation.md]: then we need to change the preceding paragraph to something like:
> For those reasons, be cautious when adding `@doc false` to functions, **use one of these two options**:

:

Comment (gabrieltaylor) [inline lib/elixir/pages/Writing Documentation.md]: Agreed. Will incorporate feedback.

File: lib/elixir/pages/Writing Documentation.md (hunk 6 above)
Comment (gabrieltaylor) [inline lib/elixir/pages/Writing Documentation.md]: Yep. Will incorporate

File: lib/elixir/pages/Writing Documentation.md (hunk 3 above)
Comment (gabrieltaylor) [inline lib/elixir/pages/Writing Documentation.md]: Good points. Will incorporate

File: lib/elixir/pages/Writing Documentation.md (hunk 1 above)
Comment (gabrieltaylor) [inline lib/elixir/pages/Writing Documentation.md]: Good feedback. Will reinstate.

File: lib/elixir/pages/Writing Documentation.md (hunk 2 above)
Comment (gabrieltaylor) [inline lib/elixir/pages/Writing Documentation.md]: I respectfully disagree. The original sentence is "fluffy". I prefer getting straight to the point.

Comment (ericmj) [inline lib/elixir/pages/Writing Documentation.md]: I prefer the shorter version. As we say on the line below: "Keep the first paragraph of the documentation concise and simple".

**Comment by lexmag** [comment]
👍

**Comment by whatyouhide** [comment]
Thanks a lot @gabrieltaylor ❤️

**Pull request merged**